### PR TITLE
Adds IntRangeDecoder, LongRangeDecoder and CharRangeDecoder

### PR DESCRIPTION
**Pull request by fagossa**
Fixes the following issues:
* https://github.com/sksamuel/hoplite/issues/37
* https://github.com/sksamuel/hoplite/issues/38
* https://github.com/sksamuel/hoplite/issues/39

### CODE REVIEW

File: hoplite-core/src/main/kotlin/com/sksamuel/hoplite/decoder/ranges.kt [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    is StringNode -> RangeDecoders.charRange(node.value).map { it.valid() }
+      .getOrElse { ConfigFailure.DecodeError(node, type).invalid() }
+    else -> ConfigFailure.DecodeError(node, type).invalid()
+  }
+}
+
+object RangeDecoders {
+
+  fun intRange(value: String): Try<IntRange> {
+    return extractContents(value, "(\\d+)\\.\\.(\\d+)".toRegex())
```
Comment (fagossa) [inline hoplite-core/src/main/kotlin/com/sksamuel/hoplite/decoder/ranges.kt]: I'm not really convinced about having the regex at this level, what do you think

File: hoplite-core/src/main/kotlin/com/sksamuel/hoplite/decoder/ranges.kt [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      .flatMap { (first, second) -> Try { IntRange(first.toInt(), second.toInt()) } }
+  }
+
+  fun longRange(value: String): Try<LongRange> {
+    return extractContents(value, "(\\d+)\\.\\.(\\d+)".toRegex())
```
Comment (fagossa) [inline hoplite-core/src/main/kotlin/com/sksamuel/hoplite/decoder/ranges.kt]: same as before

File: hoplite-core/src/main/kotlin/com/sksamuel/hoplite/decoder/ranges.kt [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+      .flatMap { (first, second) -> Try { LongRange(first.toLong(), second.toLong()) } }
+  }
+
+  fun charRange(value: String): Try<CharRange> {
+    return extractContents(value, "(\\w+)\\.\\.(\\w+)".toRegex())
```
Comment (fagossa) [inline hoplite-core/src/main/kotlin/com/sksamuel/hoplite/decoder/ranges.kt]: same as before

File: hoplite-core/src/main/kotlin/com/sksamuel/hoplite/decoder/ranges.kt (hunk 1 above)
Comment (sksamuel) [reply hoplite-core/src/main/kotlin/com/sksamuel/hoplite/decoder/ranges.kt]: I'd pull it out into a val inside the object.

**Comment by sksamuel** [comment]
awesome

**Pull request merged**